### ISSUE TITLE
How to make Custom Music, Custom Background, Etc...

### ISSUE THREAD

**Issue by GuckTubeYT** [GrowtopiaNoobs/GrowtopiaServer#1065]
How to make Custom Music, Custom Background, Etc... Where i change the file?

**Comment by NotDark** [comment]
you need to make your own items.dat

**Comment by GuckTubeYT** [comment]
how to make that?

**Comment by GuckTubeYT** [comment]
how to make items.dat

**Comment by SSebia** [comment]
> How to make Custom Music, Custom Background, Etc... Where i change the file?

Im interested too how to decode items.dat

**Comment by GuckTubeYT** [comment]
me too

**Comment by MarmoriPelaa** [comment]
The items.dat format is pretty much the same in every version. There is one database that helped me decode it partially:
https://wombat.platymuus.com/growtopia/itemdb.php
It's fairly easy to add your own item via a hex editor as long as you know what you're doing.

NOTE: Custom assets (textures, music etc.) require a separate HTTP server to provide them. Custom assets are most likely required if you are going to make a custom item.

**Comment by GuckTubeYT** [comment]
> 
> 
> The items.dat format is pretty much the same in every version. There is one database that helped me decode it partially:
> https://wombat.platymuus.com/growtopia/itemdb.php
> It's fairly easy to add your own item via a hex editor as long as you know what you're doing.
> 
> NOTE: Custom assets (textures, music etc.) require a separate HTTP server to provide them. Custom assets are most likely required if you are going to make a custom item.

Thanks MarmoriPelaa, but how to edit items.dat?

**Comment by MarmoriPelaa** [comment]
@GuckTubeYT Like I said, with a hex editor.

**Comment by playingo** [comment]
or decode and encode like we can

**Comment by GuckTubeYT** [comment]
> 
> 
> or decode and encode like we can (and even add/append NEW items without modifying)

how?

**Comment by MarmoriPelaa** [comment]
Why make me repeat myself? I just told you how: with a hex editor. What's so hard to understand!?

**Comment by GuckTubeYT** [comment]
sorry, i am noob coding, and i just can batch file

**Comment by IndogLelele** [comment]
lolololol

**Comment by GuckTubeYT** [comment]
ok now i want to edit music lobby world, how to do that?

**Comment by atomiclucky** [comment]
yeah how to decode the name then?

**Comment by MarmoriPelaa** [comment]
You don't. Put the original name in the file and edit as necessary. That's how I do it

**Comment by GuckTubeYT** [comment]
Where i do put? Can you screenshot?

**Comment by atomiclucky** [comment]
yeah, but if you want to change the name?

**Comment by GuckTubeYT** [comment]
How to Custom Block? Do I Convert PNG to HEX?

**Comment by playingo** [comment]
im never going to tell how custom items work

**Comment by atomiclucky** [comment]
@GuckTubeYT no lol?? you convert it to rttex and then type the location of .rttex to the items.dat

**Comment by Zlarex** [comment]
You open the recording system problem (not too difficult). But if you switch to this (change the data file which is not normal file (txt) and need need another way like decode the content with hex editor), you will be very confused.

You can easily do this if you have prior knowledge from the basic stage.

**Comment by GuckTubeYT** [comment]
how custom items work? how items.dat work?

**Comment by Zlarex** [comment]
What work

**Comment by GuckTubeYT** [comment]
how to edit items.dat without hex?

**Comment by GuckTubeYT** [comment]
> 
> 
> What work

custom items

**Comment by Zlarex** [comment]
Do you know how does packet system work in server, just send it from server with some code of packet to user

Maybe just hex

**Comment by MarmoriPelaa** [comment]
@GuckTubeYT You MUST use a hex editor. If you don't use a hex editor you will not get custom items, capiche?

**Comment by playingo** [comment]
@MarmoriPelaa False, we are using (secretfilename).bin for decoded content and we can encode it (with new item ids/items and not using any edited items) easily.

**Comment by playingo** [comment]
Our private server is downloadable for android om growtopia3.com/gtv3.hosts and has custom items! check it out urself

**Comment by GuckTubeYT** [comment]
> 
> 
> @MarmoriPelaa False, we are using (secretfilename).bin for decoded content and we can encode it (with new item ids/items and not using any edited items) easily. (tools are not being published and we wont tell in any kind how it works but every human with good sense should know there are 2 proper ways to do it.)

please how?

**Comment by GuckTubeYT** [comment]
bruh so hard

**Comment by GuckTubeYT** [comment]
pls help me

**Comment by GuckTubeYT** [comment]
> 
> 
> @MarmoriPelaa False, we are using (secretfilename).bin for decoded content and we can encode it (with new item ids/items and not using any edited items) easily. (tools are not being published and we wont tell in any kind how it works but every human with good sense should know there are 2 proper ways to do it.)

please how

**Comment by GuckTubeYT** [comment]
‍

**Comment by playingo** [comment]
i said we wont reveal how but anybody has it too and sold it to vyte for x euros

**Comment by GuckTubeYT** [comment]
Why?

**Comment by playingo** [comment]
because its unfair to do it. we did an incredible hard work to code it / find out how it works and people who cannot code will just get it for free thats why we dont share it.

**Comment by GuckTubeYT** [comment]
why? share is good

**Comment by MarmoriPelaa** [comment]
@playingo I do get your point on working hard on it. However, you are violating the licence. I know you're a criminal and wouldn't care less while in your approx. 16-30's and all but in a serious manner, breaking the copyright law, even on a reverse engineering related project that is meant for non-profit usage and supported by contributions of a community, is a far more serious crime than the reverse engineering itself.

As the Afferno General Public Licence states, you may distribute your server to the public, as a service or binary form, as long as you also provide the corresponding source code of it to the public.

If I were you, I would rather just politely ask for a licence change which is very unlikely to happen, or recommend some kind of a solution to make it fair for everyone.

GitHub is NOT a forum meant for fights nor is it made for selling source code that should be available to anyone, it's a site that makes collaborative programming projects a reality. The issues tab is not a trash talk sub-forum where you can show what you've made while insulting the others just because they have a different background in programming. 

Just another statement about the licence:
Even if you wanted to make another Growtopia Server from "scratch", you can't. You will always be using research from this project. There is currently no proof of your own research being made before the initial public release of the software. This means that each time you use the research from this project, you have to include the corresponding source.

**Comment by playingo** [comment]
Thats exactly true. But guess what, the items.dat decoder / encoder is a seperate program and has nothing to do with the server itself

**Comment by GuckTubeYT** [comment]
can you make tutorial?

**Comment by MarmoriPelaa** [comment]
TheLeftExit initially started the idea of decoding items.dat in practice by finding an old file from an old version of GT on android. This was followed by my poor, quick assumptions on the what the values seen in the items.dat correspond to. Then iProgramMC mentioned wombat database which was the last known key component for decoding items.dat. Everyone of us were like a small community that was working together in order to create an items.dat decoder. You collected the information and wrote a decoder intended for private use, based on our research without crediting us in any way.
So, I request you to give us credit for working on the items.dat decoder too.
For example this could be accessed on a credits dialog accessible on the Gazette:

```
...
Credits to the people working on items.dat decoder:
TheLeftExit
iProgramMC 
playingo
MarmoriPelaa
... 
```

And I think that'll be fine. 

And before you begin complaining:
No, the items.dat decoder is not fully yours even if you wrote it into a program form. It is the product of a group of collaborative people, not just you.

**Comment by playingo** [comment]
Yes but I think neither you or TheLeftExit were responsible for the help of the decoder. Firstly, anybody has this (and sold this to vyte for a 3 digits price tag) and secondly we made it for free without using any of TheLeftExit's help. We used the old wombat site of growtopia which has like alot of the variables already. For the name encoding/decoding, ness was the one responsible and for the rest of the decoder/encoder it was we both (iProgramInCpp and me) who did it. I would say it is not a work you were responsible for, but more like iProgramInCpp and ness.

**Comment by GuckTubeYT** [comment]
Bruh, i still confused
